Skatepark: Add comment styles

mikachan opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:
This PR adds styles for the comment form and comments list on Skatepark.

Comments List:

| Logged In      | Logged Out |
| ----------- | ----------- |
| ![logged-in-comments](https://user-images.githubusercontent.com/1645628/128363222-ec10e2cc-0a3b-4aa7-98ed-49111bb3a779.png)      |  ![logged-out-comments](https://user-images.githubusercontent.com/1645628/128363594-7f90db0c-3ec0-4425-8650-0b1a6c93b2d5.png)    |

Comments Form:

| Logged In      | Logged Out |
| ----------- | ----------- |
|  ![logged-in-form](https://user-images.githubusercontent.com/1645628/128368633-a0a89d79-878b-4b45-97db-da47233ad63e.png)  |  ![logged-out-form](https://user-images.githubusercontent.com/1645628/128368652-675960bf-ba4d-4902-b33c-c56a0356db13.png) |

#### Text underneath 'Leave a Reply' / 'Your email address will not be published. Required fields are marked *':
I've left this in as I thought the information was important/needed, but it's not in the design. Let me know if this should be removed!

#### Avatars:
**Size**
In the designs, the avatars are set to 72px x 72px but the default size is set to 32px x 32px. I'm guessing this needs to be changed but I'm not sure how best to do it in a block-based theme. I couldn't find any similar examples from recent themes. Would appreciate any pointers!

**Duotone**
~~I haven't applied the duotone filter to the avatars as there's [a separate issue](https://github.com/Automattic/themes/issues/4329) for adding duotone support to Skatepark. Is this OK or should I include maybe a default setting in this PR?~~

#### Related issue(s):
Closes #4317.

mikachan committed:
Skatepark: Add initial post-comments.scss

mikachan committed:
Merge branch 'trunk' into add/skatepark-comment-form-styles

# Conflicts:
#	skatepark/assets/theme.css
#	skatepark/sass/theme.scss

mikachan committed:
Skatepark: Add underline to comment meta links

mikachan committed:
Skatepark: Add comment font & spacing styles

mikachan committed:
Skatepark: Adjust comment checkbox styling

mikachan committed:
Skatepark: Adjust spacing for comment meta & paragraphs

mikachan committed:
Skatepark: Re-order comment form

mikachan committed:
Skatepark: Tweak grid column gap

mikachan committed:
Skatepark: Hide says span in comments

mikachan committed:
Skatepark: Tweak spacing for comment form

MaggieCabrera commented:
It's fine to leave the duotone for the moment, I was working on it and I found a bug

mikachan commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,169 @@
+		margin-bottom: var(--wp--custom--margin--vertical);
+
+		a {
+			--wp--custom--button--typography--font-size: var(--wp--preset--font-size--normal);
```
This is currently the wrong font size on this branch, and `-wp--preset--font-size--small` is correct. But I noticed these are about to be renamed, so I think 'normal' will be correct after that change.

MaggieCabrera replied:
Yep, normal is the correct one, I just merged that branch so you can rebase to get that change

melchoyce commented:
Hey, this is look great! Some small notes:

## Comment form
- Can we add .1em letter-spacing to the labels?
- On the logged-out comment form, can we have 45px of space between the "save my info" checkbox and the submit button instead of 30px?

## Comment list
- Can the date/edit links be medium (500) instead of regular weight?
- Let me know if it's not possible to use a bigger avatar, and I'll maybe tweak some spacing.
- Noticed the "awaiting moderation" view is a little messed up:
<img width="737" alt="Screen Shot 2021-08-05 at 4 21 58 PM" src="https://user-images.githubusercontent.com/2846578/128416132-95074bb7-8b73-4b3d-8702-5f762a5e8b5c.png">

^ We might want to test that in Blockbase & Quadrat to see if that's a problem for those themes as well

MaggieCabrera commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,169 @@
+
+		input:not([type=submit]):not([type=checkbox]),
+		textarea {
+			border: 3px solid var(--wp--custom--color--primary);
```
I think these should be picking up from theme.json (`custom.form.border`) so that all form-related stuff is consistent. In fact, that should be on blockbase too.

MaggieCabrera commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,169 @@
+					position: absolute;
+					top: -0.4em;
+					left: 0;
+					width: 30px;
+					height: 30px;
+					border: 3px solid var(--wp--custom--color--primary);
+				}
```
I see this is also on Quadrat, I'd say this should be on Blockbase too, but I'd appreciate other team member's opinions. Ideally we should add a checkbox inside theme.json's `custom.form` section. I feel strongly that we need consistent form styles defined for the theme from theme.json and reused whenever needed. This will be helpful when we style things like WooCommerce's forms, for example. A mixin for checkboxes like we have for buttons could probably be useful if I'm thinking ahead.

scruffian replied:
I agree, this should be moved to Blockbase.

MaggieCabrera commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,169 @@
+		display: grid;
+		column-gap: 1.2em;
+		grid-template-rows: auto;
+		grid-template-areas:
```
<3 `grid-template-areas`

MaggieCabrera commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,169 @@
+
+.wp-block-post-comments {
+
+	line-height: var(--wp--custom--body--typography--line-height);
```
if this is not the case by default, this is a good thing to do from Blockbase too

mikachan replied:
Looks like this is the case by default actually, I've just tried removing it and it made no difference. It looks like it's applied to the body in the inline styles at the top of the page (the 'global-styles-inline-css' block).

MaggieCabrera commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,169 @@
+	}
+
+	.commentlist {
+		margin-bottom: 60px;
```
this could be something like `calc( 3 * var(--wp--custom--margin--baseline) )`

mikachan committed:
Merge branch 'trunk' into add/skatepark-comment-form-styles

mikachan committed:
Skatepark: Use calc for commetlist bottom margin

mikachan committed:
Skatepark: Remove line-height from comments block

Not required - the same is applied via global styles

mikachan committed:
Skatepark: Add custom.form.border to child-theme.json

mikachan committed:
Skatepark: Add custom.form.label styles to child-theme.json

mikachan committed:
Skatepark: Change bottom margin on cookies consent container

mikachan committed:
Skatepark: Increase font weight of comment meta links

mikachan committed:
Blockbase: Improve comment awaiting moderation styling

mikachan committed:
Blockbase: Add comment form input & textarea border styles

mikachan commented:
Thanks so much everyone for all the comments! I think I've addressed everything apart from moving the checkbox styles into Blockbase. I've started this but I felt like I was making this PR too big by including it here, so I'm thinking of making a separate issue.

@melchoyce: Sorry, I feel like a lot of your comments were me missing things from the design, gah. I've added each thing you've mentioned, other than increasing the avatar size*. 

> * Noticed the "awaiting moderation" view is a little messed up

Great spot! You're right, this also seems to be an issue in Blockbase and Quadrat as well, so I've added a fix for this in Blockbase.

Example in Quadrat before fix:
![moderation-problem-quadrat](https://user-images.githubusercontent.com/1645628/128520717-28d8cb14-b05c-4ea1-9111-fb0f3ea28691.png)

Skatepark with fix:
![comment-awaiting-moderation](https://user-images.githubusercontent.com/1645628/128520784-13f333e5-ffbb-4820-a7f0-7847b61d35ff.png)

*To everyone: is there a standard way to increase the size of the avatars in the comments block? I know there's a way to do it within a PHP template (e.g. comments.php), but I'm not sure where to edit this as a block and I can't find any similar examples.

MaggieCabrera commented:
> *To everyone: is there a standard way to increase the size of the avatars in the comments block? I know there's a way to do it within a PHP template (e.g. comments.php), but I'm not sure where to edit this as a block and I can't find any similar examples.

The comments block right now is very basic and needs a lot of work, at first glance I think the avatar size is hardcoded, but it would make sense that in the future the block provided means to alter the size of it.

> I've started this but I felt like I was making this PR too big by including it here, so I'm thinking of making a separate issue.

I don't think you need to make a new one, it's easier to work on just one PR instead of two that depend on each other. The code won't get bigger, it will just move from skatepark to Blockbase (and deleted from Quadrat, sure). If you think it will help you, go ahead, but I don't see an issue in reviewing everything together.

mikachan commented:
> The comments block right now is very basic and needs a lot of work, at first glance I think the avatar size is hardcoded, but it would make sense that in the future the block provided means to alter the size of it.

Ah I see, thanks for confirming. Maybe for now it's best for @melchoyce to tweak some spacing to work with the hardcoded size.

> I don't think you need to make a new one, it's easier to work on just one PR instead of two that depend on each other.

Great OK, I'll carry on in here then. I wasn't sure what the best thing to do was! Thanks!

mikachan committed:
Blockbase: Add comment form checkbox styling

Including updates to Quadrat, Seedlet Blocks & Skatepark

mikachan committed:
Merge branch 'trunk' into add/skatepark-comment-form-styles

# Conflicts:
#	skatepark/assets/theme.css
#	skatepark/sass/theme.scss
#	skatepark/theme.json

MaggieCabrera commented on blockbase/sass/blocks/_post-comments.scss:
```diff
@@ -12,14 +12,52 @@
+			input[type="checkbox"] {
+				margin-left: 0;
+				
+				& + ::before,
+				& + ::after {
+					box-sizing: border-box;
```
These are working for Quadrat but not for Blockbase because of the missing `position:relative;`. I think they need to be `& + label::before` and we need to add `position:relative;` to `& + label`

MaggieCabrera commented on blockbase/sass/blocks/_post-comments.scss:
```diff
@@ -12,14 +12,52 @@
+		}
+
+		.comment-form-cookies-consent {
+			input[type="checkbox"] {
```
This needs the `@include hide-accesibly;` mixin to hide the original checkbox too

MaggieCabrera commented on quadrat/theme.json:
```diff
@@ -217,6 +217,31 @@
+				},
+				"checkbox": {
+					"checked": {
+						"content": "\"\\2715\"",
```
Smart!

MaggieCabrera commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,142 @@
+
+		input:not([type=submit]):not([type=checkbox]),
+		textarea {
+			background: var(--wp--custom--color--background);
```
I think the background is already defined on Blockbase

MaggieCabrera commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,142 @@
+		column-gap: 1.2em;
+		grid-template-rows: auto;
+		grid-template-areas:
+			"notes notes"
+			"comment comment"
+			"author email"
+			"url url"
+			"cookies-consent cookies-consent"
+			"form-submit form-submit";
```
I just noticed that this is repeated on Quadrat. I think it's safe to handle this in Blockbase too and repeat less code over the child themes, since they are mostly going to have the same layout, what do you think @kjellr?

kjellr replied:
Yeah, definitely. I expect this comment form layout will be mostly identical across our themes.

MaggieCabrera commented on skatepark/sass/blocks/_post-comments.scss:
```diff
@@ -0,0 +1,142 @@
+					margin-left: 0;
+					padding-left: 3em;
+					position: relative;
+				}
```
yeah, this looks like it has to be done on the parent theme too, this is what's missing on Blockbase and why it's not working on it

MaggieCabrera commented on quadrat/theme.json:
```diff
@@ -217,6 +217,31 @@
 				"typography": {
 					"fontSize": "var(--wp--preset--font-size--normal)"
+				},
+				"checkbox": {
```
Are we defining these only on the child themes? these should come from Blockbase and then rebuilt into the children

MaggieCabrera replied:
And adapted when needed ofc

mikachan replied:
The reason I skipped Blockbase (and didn't move the other checkbox styles) is because it looks like Blockbase is using a native checkbox:

![native checkbox](https://user-images.githubusercontent.com/1645628/129063930-47e525c0-dd3a-4208-8d0b-122ede61cf90.png)

This is what it looks like when I move all these custom checkbox related styles into Blockbase:

![custom checkbox](https://user-images.githubusercontent.com/1645628/129063988-d8a859cd-046d-4208-9c1e-4072420e4cc2.png)

If we ideally want all Blockbase and children using a custom checkbox, should I make the border of the custom Blockbase one any darker, similar to the native one? My screenshots are from Firefox. Mainly asking as the box is so tiny and I'm not sure the light grey is great for accessibility. It's using the same border colour as the other form inputs too, so then I don't know if they should all change or just the checkbox border. What do you think?

MaggieCabrera replied:
Yeah, that makes sense. Blockbase styles are based off of Blank Canvas', which looks the same on Firefox (Chrome shows a very small checkbox with a tiny border). In turn Blank Canvas is a child of Seedlet:

Seedlet | Blank Canvas
--- | ---
<img width="756" alt="Screenshot 2021-08-11 at 18 34 11" src="https://user-images.githubusercontent.com/3593343/129068152-5e68250d-7098-47fe-a975-7766b1fb3005.png"> | <img width="814" alt="Screenshot 2021-08-11 at 18 33 46" src="https://user-images.githubusercontent.com/3593343/129068160-76e2793b-37f2-4ff9-a88f-105eac8a1070.png">

I think this is a minor change and I'd rather remove as much of the CSS from the children as possible. We could make the custom checkbox on Blockbase look the same color as it does on Firefox without being the native one but using our theme.json settings, or maybe we could change all the inputs borders on Blockbase a more contrasting color. @kjellr what do you think about this?

kjellr replied:
I'm not sure I follow. 😅 What are the two options here?

MaggieCabrera replied:
so right now Blockbase (and Blank Canvas and Seedlet) have native checkboxes for the comments form. The idea is that since Quadrat and Skatepark used a styled one, have that code be in Blockbase too, and make it custom styled instead of native, but that brings the a11y issues @mikachan pointed out. I don't have a preference as to how to prevent that, besides wanting to avoid the code duplication of having the child themes all provide the custom css for the checkboxes instead of providing it from Blockbase.

kjellr replied:
Thanks for clarification. I think it's fine to move the styles to Blockbase, but we should make sure that all form fields are made darker so as to avoid the a11y issue. (It looks like the checkbox border is just as light as the other form field borders, which are presumably inaccessible as well). 

Since that's more of a global change, I think it would be fine to explore it in a followup PR.

mikachan replied:
Thanks @kjellr, that makes sense, I'll do that! Thanks @MaggieCabrera for clarifying.

mikachan committed:
Blockbase: Move label position relative to Blockbase

mikachan committed:
Blockbase: Move hide-accesibly mixin to Blockbase

mikachan committed:
Blockbase: Move textarea background to Blockbase

mikachan committed:
Blockbase: Move custom checkbox styles to Blockbase theme.json

mikachan committed:
Blockbase: Move form grid styles to Blockbase

MaggieCabrera committed:
moved mixins include on BB

MaggieCabrera commented:
I'm bringing this in, thank you @mikachan for your hard work on this, it's looking awesome!
